File: The-Golden-Leaf-server/src/main/java/com/example/datban/repository/DatBanRepository.java
package com.example.datban.repository;

import com.example.datban.model.DatBan;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import java.util.Optional; // <-- thêm dòng này

@Repository
public interface DatBanRepository extends JpaRepository<DatBan, Long> {
    DatBan findTopByEmailOrderByIdDatDesc(String email);
    Optional<DatBan> findTopByOrderByIdDatDesc(); // Lấy DatBan mới nhất
}

File: Datban/app/src/main/java/com/example/giaodien/data/repository/DatBanRepository.kt
package com.example.giaodien.data.repository

import com.example.giaodien.data.model.DatBan
import com.example.giaodien.data.network.RetrofitInstance

class DatBanRepository {

    suspend fun datBan(datBan: DatBan): DatBan {
        return RetrofitInstance.api.createDatBan(datBan)
    }
    // ✅ THÊM HÀM MỚI: Lấy DatBan mới nhất từ Server
    suspend fun getLatestDatBan(): DatBan {
        // Hàm này tự động gửi email qua token nhờ AuthInterceptor
        return RetrofitInstance.api.getLatestDatBan()
    }
}
